### DISCUSSION TITLE
Peak annotation

### DISCUSSION THREAD

**Discussion by l-magnificence** [stuart-lab/signac#423] [Q&A] [answered]
Can we get information including peak annotation from signac that like what ArchR did?
This information annotated peaks based on their genomic context as promoters (-2,000/+100 bp from a TSS), exonic, intronic or intergenic based on UCSC annotations and also contained the peak-associated gene with biotypes such as protein-coding, lncRNA , small RNA and other.
![image](https://user-images.githubusercontent.com/68798288/112273671-906f0c80-8cb8-11eb-937c-4248b88cb3f6.png)

**Comment by timoast** [comment, marked as answer]
You can use functions in the `GenomicRanges` package for this. For example, you can find the closes annotated feature and the distance to the feature using the `distanceToNearest()` function, eg:

```r
distanceToNearest(atac_small, subject = Annotation(atac_small))
```

See [here](https://satijalab.org/signac/articles/data_structures.html#other-chromatinassay-methods) for more information. Alternatively you can use the [`ClosestFeature()`](https://satijalab.org/signac/reference/closestfeature) function.

**Reply by kaibtan** [reply]
Hi @timoast thanks for the explanation.

Using the ClosestFeature() with the peaks annotated by Ensembl, I managed to get gene biotypes (protein-coding, lincRNA etc) and type (exon, utr, cds, gap)
`annotation <- GetGRangesFromEnsDb(ensdb = EnsDb.Mmusculus.v79)`

Q1: What does "gap" mean? 

Q2: While gene biotypes are informative, how can we get more granular details like intron, intergenic, 3' UTR, 5'UTR, and promoter, as [this tutorial did using ChIPseeker](https://bioconductor.org/packages/devel/bioc/vignettes/ChIPseeker/inst/doc/ChIPseeker.html#peak-annotation)

Q3: I came across [TxDb.Mmusculus.UCSC.mm10.knownGene](https://www.bioconductor.org/packages/devel/workflows/vignettes/annotation/inst/doc/Annotating_Genomic_Ranges.html#mouse-mm10). Is there a way to incorporate TxDb parameters into an EnsDb GRanges object, which can be used for peak annotation? 

Many thanks for your help in advance

**Reply by RaghadShu** [reply]
No answers to specifically q2?..
